wsagent/che-core-api-project/src/test/java/org/eclipse/che/api/project/server/ProjectServiceTest.java

```diff
@@ -928,6 +939,50 @@ public void testDeleteProject() throws Exception {
     }
 
     @Test
+    public void testDeleteProjectsConcurrently() throws Exception {
+        int threadNumber = 100;
+        ExecutorService executor = Executors.newFixedThreadPool(threadNumber);
+        CountDownLatch countDownLatch = new CountDownLatch(threadNumber);
+        List<Future<ContainerResponse>> futures = new LinkedList<>();
+
+
+        for (int i = 0; i < threadNumber; i++) {
+            addMockedProjectConfigDto(ptRegistry.getProjectType("my_project_type"), "my_project_name" + i);
+        }
+
+        IntStream.range(0, threadNumber).forEach(
```

Comment:
Exactly! As we can see in [this article](https://jaxenter.com/java-performance-tutorial-how-fast-are-the-java-8-streams-118830.html) in general cases java streams are a bit slower than `ArrayList`, so I would obviously use simple java loop here because it would also improve readability but the limitation is that it is not possible to refer to a non final variable.